### ISSUE TITLE
Conditional redefine of MPIFC is broken in maint-1.0

### ISSUE THREAD

**Issue by rljacob** [E3SM-Project/E3SM#3027]
If you try to redefine the mpi compiler wrappers with something like:
`<MPIFC MPILIB="impi"> mpiifort </MPIFC>`
You'll get a compile line that starts like this:
`mpif90  mpiifort <options>  file.F90`

This problem doesn't happen on master.

The config_compilers.xml in maint-1.0 doesn't use that construction in any definitions so we didn't notice it until we tried (on compy).

**Comment by rljacob** [comment]
The source of the problem was found by @singhbalwinder in cime:
```
if output_format == "make":
                if key.startswith("ADD_"):
                    fd.write("{} {} b1+= {}\n".format(" " * width, key[4:], value))
                else:
                    fd.write("{} {} += {}\n".format(" " * width, key, value))
```

The last line should have `:=` instead of `+=`.  Making that changes allows the construction above to work.

**Comment by rljacob** [comment]
Fixed in #3023
